features/netsocket/NetworkInterface.h

```diff
@@ -100,6 +100,24 @@ class NetworkInterface: public DNS {
      */
     virtual const char *get_mac_address();
 
+    /** Set the MAC address to the interface.
+     *
+     *  Provided MAC address is set the the network interface. Address must be
+     *  set before calling the interface connect() method. EUI-48 MAC addresses
```

Comment:
My only question here is should we mention that this must be unique globally and there are places where to get these and also that some interfaces already have an unique MAC address written in HW and this should not be called on those interfaces

Sort of own chapter on the possible hazards if this API is called without proper knowledge

Reply:
Method description updated, please re-review.